### DISCUSSION TITLE
inserting new entity into metadata storage in Mikro ORM instance

### DISCUSSION THREAD

**Discussion by hubbusysyuhada** [mikro-orm/mikro-orm#2262] [Q&A] [answered]
Hi, currently I'm working on a project that requires me to use a dynamic database structure with 0 entities given at mikro orm instance init.

In the middle of process, I create a table in my database which name, column types, and default value could be vary. I use `EntityGenerator.generate()` to generate my latest database schema and save it into my project directory every time there's DDL involved.

My question is, how can I insert this new entity files that has just been generated into the running mikro orm connection so I could use it in my EntityManager? I tried to close the connection with `orm.close()` and initialized it again, but it seems like not working as the integration test I use left hanging

**Comment by B4nan** [comment, marked as answer]
In v5 there is [`orm.discoverEntity()`](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/MikroORM.ts#L117) helper method for dynamic registration of new entities, it will basically load the metadata on the fly, no reconnection needed. You can probably do the same in v4 too, but it will be a bit more code (e.g. you will need to create new instance of `MetadataDiscovery` is that is not exposed in v4).

**Reply by hubbusysyuhada** [reply]
many thanks!
